RecoEgamma/EgammaElectronProducers/src/LowPtGsfElectronFeatures.cc

```diff
@@ -164,6 +164,400 @@ namespace lowptgsfeleid {
+          sc_clus2_deta = -999.;
+          sc_clus1_E = -999.;
+          sc_clus2_E = -999.;
+          sc_clus1_E_ov_p = -999.;
+          sc_clus2_E_ov_p = -999.;
+
+          // track-clusters match
+          clusNum = 0;
+          try {
+            if (sc->clustersSize() > 0 && sc->clustersBegin() != sc->clustersEnd()) {
+              for (auto& cluster : sc->clusters()) {
+                double pi_ = 3.1415926535;
+                float deta = std::fabs(ecal_pos.eta() - cluster->eta());
+                float dphi = std::fabs(ecal_pos.phi() - cluster->phi());
+                if (dphi > pi_)
+                  dphi -= 2 * pi_;
+                if (ecal_pos.phi() - cluster->phi() < 0)
+                  dphi = -dphi;
+                if (ecal_pos.eta() - cluster->eta() < 0)
+                  deta = -deta;
+
+                if (clusNum == i1) {
+                  sc_clus1_E = cluster->energy();
+                  if (gsf->pMode() > 0)
+                    sc_clus1_E_ov_p = cluster->energy() / gsf->pMode();
+                  sc_clus1_nxtal = (int)cluster->size();
+                  if (reach_ECAL > 0) {
+                    sc_clus1_deta = deta;
+                    sc_clus1_dphi = dphi;
+                  }
+                } else if (clusNum == i2) {
+                  sc_clus2_E = cluster->energy();
+                  if (gsf->pMode() > 0)
+                    sc_clus2_E_ov_p = cluster->energy() / gsf->pMode();
+                  if (reach_ECAL > 0) {
+                    sc_clus2_deta = deta;
+                    sc_clus2_dphi = dphi;
+                  }
+                }
+                clusNum++;
+              }
+            }
+          } catch (...) {
+            std::cout << "caught an exception" << std::endl;
+          }
+        }
+      }
+    }  // clusters
+
+    // Out-of-range
+    if (eid_rho < 0)
+      eid_rho = 0;
+    if (eid_rho > 100)
+      eid_rho = 100;
+    if (eid_sc_eta < -5)
+      eid_sc_eta = -5;
+    if (eid_sc_eta > 5)
+      eid_sc_eta = 5;
+    if (eid_shape_full5x5_r9 < 0)
+      eid_shape_full5x5_r9 = 0;
+    if (eid_shape_full5x5_r9 > 2)
+      eid_shape_full5x5_r9 = 2;
+    if (eid_sc_etaWidth < 0)
+      eid_sc_etaWidth = 0;
+    if (eid_sc_etaWidth > 3.14)
+      eid_sc_etaWidth = 3.14;
+    if (eid_sc_phiWidth < 0)
+      eid_sc_phiWidth = 0;
+    if (eid_sc_phiWidth > 3.14)
+      eid_sc_phiWidth = 3.14;
+    if (eid_shape_full5x5_HoverE < 0)
+      eid_shape_full5x5_HoverE = 0;
+    if (eid_shape_full5x5_HoverE > 50)
+      eid_shape_full5x5_HoverE = 50;
+    if (eid_trk_nhits < -1)
+      eid_trk_nhits = -1;
+    if (eid_trk_nhits > 50)
+      eid_trk_nhits = 50;
+    if (eid_trk_chi2red < -1)
+      eid_trk_chi2red = -1;
+    if (eid_trk_chi2red > 50)
+      eid_trk_chi2red = 50;
+    if (eid_gsf_chi2red < -1)
+      eid_gsf_chi2red = -1;
+    if (eid_gsf_chi2red > 100)
+      eid_gsf_chi2red = 100;
+    if (eid_brem_frac < 0)
+      eid_brem_frac = -1;
```

Comment:
is this intentionally set to -1? unlike other cases this is not a simple clamp/truncation

Reply:
I think yes. @crovelli, please can you confirm?

Reply:
yes, it's intentional